BranchMetrics/AdobeBranchExtension-Android
# [SDK-1911][major] Migrated to AEPCore

## Reference
SDK-1911 -- [Android] Update AdobeLaunchExtensions to support MobileCore versions 2+ 

## Summary
This PR reflects updates to the Adobe SDK, which has transitioned from the Adobe Cloud Platform (ACP) to the Adobe Experience Platform (AEP). This transition from ACP to AEP involves a major update resulting in changes to several class and method names and definitions. This PR also includes updating the Branch SDK from 5.2.7 to 5.4.0.

**[Link to Adobe's documentation on the changes can be found here ](https://developer.adobe.com/client-sdks/previous-versions/documentation/migrate-to-android/)**

#### Key changes include:
- Changes in the way listeners are created
- Changes in the way extensions are registered and initialized
- Slight changes to several of the method definitions (`getSharedState`, `registerEventListener`, `createSharedState`)
- Removal of `ExtensionErrorCallback` and `ExtensionError`
- New `SharedStateResult` object
- New, easier way to get Adobe context

## Motivation

Clients who want to take advantage of the new AEP SDK's requested that we update our extension to be compatible.

## Type Of Change


## Testing Instructions

**To verify these changes:**
1. Check out the repository and integrate it into a sample app or use the existing AdobeBranchExample app.
2. Ensure that all events and Adobe IDs are correctly passed through to Branch.
3. Verify that all of the previous functionalities of the extension remain intact.









cc @BranchMetrics/saas-sdk-devs for visibility.
